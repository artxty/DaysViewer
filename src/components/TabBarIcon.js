import React from 'react';
import PropTypes from 'prop-types';
import { Ionicons } from '@expo/vector-icons';

import Colors from '../constants/Colors';

export default function TabBarIcon({ name, focused }) {
  return (
    <Ionicons
      name={name}
      size={26}
      style={{ marginBottom: -3 }}
      color={focused ? Colors.tabIconSelected : Colors.tabIconDefault}
    />
  );
}

TabBarIcon.propTypes = {
  name: PropTypes.string.isRequired,
  focused: PropTypes.bool.isRequired,
};
