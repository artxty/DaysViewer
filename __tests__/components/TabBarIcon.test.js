import React from 'react';
import renderer from 'react-test-renderer';

import TabBarIcon from '../../src/components/TabBarIcon';

jest.mock('@expo/vector-icons', () => ({
  Ionicons: 'Ionicons',
}));

it('renders correctly', () => {
  const tree = renderer.create(<TabBarIcon name="test" focused={false} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders correctly when focused', () => {
  const tree = renderer.create(<TabBarIcon name="test" focused />);

  expect(tree.toJSON()).toMatchSnapshot();
});
