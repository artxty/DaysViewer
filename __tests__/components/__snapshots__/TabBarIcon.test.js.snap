// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Ionicons
  color="#ccc"
  name="test"
  size={26}
  style={
    Object {
      "marginBottom": -3,
    }
  }
/>
`;

exports[`renders correctly when focused 1`] = `
<Ionicons
  color="#2f95dc"
  name="test"
  size={26}
  style={
    Object {
      "marginBottom": -3,
    }
  }
/>
`;
